gradle-metric-schema/src/main/java/com/palantir/metric/schema/gradle/MetricSchemaPlugin.java

```diff
@@ -71,6 +72,16 @@ public void apply(Project project) {
                         .apply(MetricSchemaMarkdownPlugin.class));
     }
 
+    private static void configureDependentTasks(Project project, TaskProvider<? extends Task> task) {
+        SourceSet mainSourceSet = getMainSourceSet(project);
+        project.getTasks()
+                .named(mainSourceSet.getCompileJavaTaskName())
+                .configure(compileJava -> compileJava.dependsOn(task));
+        project.getTasks()
+                .named(mainSourceSet.getSourcesJarTaskName())
+                .configure(sourcesJar -> sourcesJar.dependsOn(task));
+    }
```

Comment:
Instead of explicitly configuring task dependencies, I think we want to do something along these lines (and similar for the `resources` piece as well):
```groovy
getMainSourceSet(project).getJava().srcDir(taskProviderWithSrcDirDeclaredOutputs);
```

The TaskProvider should carry the correct dependency info and prerequisites such that gradle can build an accurate execution graph.

Reply:
Updated and tested locally on top of https://github.com/palantir/tritium/pull/1871 so that now succeeds on Gradle 8.